apps/meteor/app/livechat/server/api/v1/agent.ts

```diff
@@ -58,3 +61,49 @@ API.v1.addRoute(
 		},
 	},
 );
+
+API.v1.addRoute(
+	'livechat/agent.status',
+	{ authRequired: true, permissionsRequired: ['view-l-room'], validateParams: isPOSTLivechatAgentStatusProps },
+	{
+		async post() {
+			const { status, agentId: inputAgentId } = this.bodyParams;
+
+			const agentId = inputAgentId || this.userId;
+
+			const agent = await Users.findOneAgentById(agentId, {
```

Comment:
when using `projection` it is always good to provide a generic with the returned fields, it adds an extra layer of type safety:
```suggestion
			const agent = await Users.findOneAgentById<Pick<IUser, 'status' | 'statusLivechat'>>(agentId, {
```